src/services/blocks/BlocksService.ts

```diff
@@ -216,12 +227,62 @@ export class BlocksService extends AbstractService {
 					continue;
 				}
 
+				// The Dispatch class used to key into `blockWeights.perClass`
+				// We set default to be normal.
+				let weightInfoClass = 'normal';
+
+				if (weightInfo.class.isMandatory) {
+					weightInfoClass = 'mandatory';
+				} else if (weightInfo.class.isOperational) {
+					weightInfoClass = 'operational';
+				}
+
+				/**
+				 * RESOURCES
+				 * This `extrinsicBaseWeight` changed from using system.extrinsicBaseWeight => system.blockWeights.perClass[weightInfoClass].baseExtrinsic
+				 * in polkadot v0.8.27 due to this pr: https://github.com/paritytech/substrate/pull/6629 .
+				 * https://github.com/paritytech/substrate-api-sidecar/issues/393 .
+				 * https://github.com/polkadot-js/api/issues/2365
+				 */
+				let extrinsicBaseWeight;
+				if (
+					specName !== api.runtimeVersion.specName.toString() ||
+					specVersion !== api.runtimeVersion.specVersion.toNumber()
+				) {
+					// We are in a runtime that does **not** match the decorated metadata in the api,
+					// so we must fetch the correct metadata, decorate it and pull out the constant
+					const decorated = expandMetadata(api.registry, metadata);
```

Comment:
As mentioned above, I think this is expensive and only needs to be done once per block so we can move it out of the loop.

Reply:
Yea as talked about earlier, i'll revert it to the original plan, and put metadata and the decorated variable back into `createCalcFee` and build some logic to stop any and all extra expensive calls.